### PR TITLE
docs: update the headless tutorial

### PR DESCRIPTION
**Pull request by salihozdemir**
Updated the headless tutorial and depend examples.

### CODE REVIEW

File: documentation/docs/tutorial/1-getting-started/headless/3-generate-crud-pages.md [hunk 1]
```diff
@@ -398,6 +499,81 @@ const App = () => {
 render(<App />);
 ```
 
+## Adding layout to the pages
+
+The Inferencer generates pages without any predefined layout, whereas creating a new app with **refine** comes with a default layout that includes basic components such as `Menu` and `Breadcrumb` to navigate between pages. Let's add a layout to our app.
```
Comment (DoguhanOzgurAkca) [inline documentation/docs/tutorial/1-getting-started/headless/3-generate-crud-pages.md:504]:
```suggestion
Though refine has a default layout with basic components such as Menu and Breadcrumb, which is used to navigate between pages, Inferencer doesn't add any layouts to the pages it creates. 
Here is how we can add the default layout in our app:
```

moving this above "previewing the auto generated pages" and adding a "I have added the default layout to my app." to the checklist down below(between the existing two to preserve the order they are in the page) would also be good.

Comment (salihozdemir) [reply documentation/docs/tutorial/1-getting-started/headless/3-generate-crud-pages.md]: This section has been removed and a new section has been added to the new unit 5.7 instead, can you check it, please?

**Review by aliemir — APPROVED** [review]

File: documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+It uses the `useBreadcrumb` hook to get the breadcrumb items and renders them as a list. Also, it uses the `Link` component from the `react-router-dom` package to render the links.
+
+> For more information, refer to the [`useBreadcrumb` documentation &#8594](/docs/api-reference/core/hooks/useBreadcrumb)
+
+<br />
+
+<Checklist>
+
+<ChecklistItem id="layout-menu-breadcrumb">
+I have learned about the layout, menu and breadcrumb components.
```
Comment (necatiozmen) [inline documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md:100]:
```suggestion
I have learned about the Layout, Menu and Breadcrumb components.
```

File: documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+};
+```
+
+It uses the `useMenu` hook to get the menu items and renders them as a list. Also, it uses the `NavLink` component from the `react-router-dom` package to render the links.
+
+> For more information, refer to the [`useMenu` documentation &#8594](/docs/api-reference/core/hooks/ui/useMenu)
+
+## Breadcrumb
+
+The `Breadcrumb` component is located in the `src/components/breadcrumb.tsx` file and it looks like this:
```
Comment (necatiozmen) [inline documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md:66]:
```suggestion
The `Breadcrumb` component is located in the `src/components/breadcrumb.tsx` file and it looks like below:
```

File: documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        </div>
+    );
+};
+```
+
+It has a `Menu` and a `Breadcrumb` component and it renders the children. You can completely customize this component to your needs.
+
+## Menu
+
+The `Menu` component is located in the `src/components/menu.tsx` file and it looks like this:
```
Comment (necatiozmen) [inline documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md:37]:
```suggestion
The `Menu` component is located in the `src/components/menu.tsx` file and it looks like below:
```

File: documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+title: 7. Layout, Menu and Breadcrumb
+tutorial:
+    order: 0
+    prev: tutorial/adding-crud-pages/{preferredUI}/adding-sort-and-filters
+    next: tutorial/understanding-authprovider/index
+---
+
+## Layout
+
+When you create a new application with the **refine**, it creates a default layout under the `src/components/layout.tsx` file and it looks like this:
```
Comment (necatiozmen) [inline documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md:12]:
```suggestion
When you create a new application with the **refine**, it creates a default layout under the `src/components/layout.tsx` file and it looks like below:
```

**Review by necatiozmen — CHANGES_REQUESTED** [review]

**Review by necatiozmen — APPROVED** [review]

**Review by alicanerdurmaz — APPROVED** [review]

**Review by aliemir — APPROVED** [review]

File: documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    order: 0
+    prev: tutorial/adding-crud-pages/{preferredUI}/adding-sort-and-filters
+    next: tutorial/understanding-authprovider/index
+---
+
+import SharedComponents from "../../../partials/tutorial/headless-layout.md";
+
+<SharedComponents />
+
+In [Unit 2.4](/docs/tutorial/getting-started/headless/generate-crud-pages/), we created the CRUD pages automatically with Inferencer, and wrapped them with a `<Layout>` component.
```
Comment (aliemir) [inline documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md:14]:
```suggestion
In [Unit 2.4](/docs/tutorial/getting-started/headless/generate-crud-pages/), we created the CRUD pages automatically with Inferencer and wrapped them with a `<Layout>` component.
```

File: documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+This component is provided when you create a new application using `create-refine-app` to help you get started quickly by providing a simple implementations of `Menu` and `Breadcrumb` components using **refine**'s hooks, [`useMenu`](/docs/api-reference/core/hooks/ui/useMenu) and [`useBreadcrumb`](/docs/api-reference/core/hooks/useBreadcrumb).
```
Comment (aliemir) [inline documentation/docs/tutorial/4-adding-crud-pages/headless/layout-menu-breadcrumb.md:16]:
```suggestion
This component is provided when you create a new application using `create-refine-app` to help you get started quickly by providing simple implementations of `Menu` and `Breadcrumb` components using **refine**'s hooks, [`useMenu`](/docs/api-reference/core/hooks/ui/useMenu) and [`useBreadcrumb`](/docs/api-reference/core/hooks/useBreadcrumb).
```

**Pull request merged**